notarization_poller/src/notarization_poller/worker.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+"""Notarization poller worker functions.
+
+Attributes:
+    log (logging.Logger): the log object for the module.
+
+"""
+import asyncio
+import logging
+import signal
+import socket
+import sys
+import typing
+from asyncio import sleep
+
+import aiohttp
+import arrow
+import taskcluster
+from taskcluster.aio import Queue
+
+from notarization_poller.config import get_config_from_cmdln, update_logging_config
+from notarization_poller.constants import MAX_CLAIM_WORK_TASKS
+from notarization_poller.task import Task
+from scriptworker_client.constants import STATUSES
+from scriptworker_client.utils import makedirs, rm
+
+log = logging.getLogger(__name__)
+
+
+# claim_work {{{1
+async def claim_work(config, worker_queue, num_tasks=1):
+    """Find and claim the next pending task(s) in the queue, if any.
+
+    Args:
+        config (dict): the running config
+
+    Returns:
+        dict: a dict containing a list of the task definitions of the tasks claimed.
+
+    """
+    log.debug("Calling claimWork for {}/{}...".format(config["worker_group"], config["worker_id"]))
+    payload = {"workerGroup": config["worker_group"], "workerId": config["worker_id"], "tasks": num_tasks}
+    try:
+        return await worker_queue.claimWork(config["provisioner_id"], config["worker_type"], payload)
+    except (taskcluster.exceptions.TaskclusterFailure, aiohttp.ClientError) as exc:
+        log.warning("{} {}".format(exc.__class__, exc))
+
+
+# RunTasks {{{1
+class RunTasks:
+    """Manages processing of Taskcluster tasks."""
+
+    def __init__(self, config):
+        """Initialize RunTasks."""
+        self.config = config
+        self.running_tasks = []
+        self.last_claim_work = arrow.get(0)
+        self.is_stopped = False
+        self.is_cancelled = False
+        self.future = None
+
+    async def invoke(self):
+        """Claims and processes Taskcluster work."""
+        while not self.is_cancelled and not self.is_stopped:
+            num_tasks_to_claim = min(self.config["max_concurrent_tasks"] - len(self.running_tasks), MAX_CLAIM_WORK_TASKS)
+            if num_tasks_to_claim > 0 and arrow.utcnow().timestamp - self.last_claim_work.timestamp >= self.config["claim_work_interval"]:
```

Comment:
Is the timestamp part of this ever going be false? We always sleep for an amount of time that will make the timestamp part true.